service/matching/poller/manager.go

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package poller
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/cache"
+	"github.com/uber/cadence/common/clock"
+	"github.com/uber/cadence/common/types"
+)
+
+const (
+	pollerHistoryInitSize    = 0
+	pollerHistoryInitMaxSize = 5000
+	pollerHistoryTTL         = 5 * time.Minute
+)
+
+type (
+	Info struct {
+		Identity       string
+		RatePerSecond  float64
+		IsolationGroup string
+	}
+
+	Manager interface {
+		StartPoll(pollerID string, cancelFunc context.CancelFunc, info *Info)
+		EndPoll(pollerID string)
+		CancelPoll(pollerID string) bool
+		HasGroupAfter(isolationGroup string, earliest time.Time) bool
+		HasAnyAfter(after time.Time) bool
```

Comment:
`HasPollerAfter` is a better name, I think

Reply:
Done